### DISCUSSION TITLE
How to set default classes with ui.markdown?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1940] [Q&A] [answered]
### Question

Hi All,

I was recently trying to learn and implement the default formatting options released recently (a big thanks for that, it's a huge help 😄). So I realized that the ```ui.markdown``` element is not behaving the same way as ```ui.label```.

Ex: This is the code I tried to compare:
```python
ui.label().default_classes('text-2xl')
ui.label('This is my sample text').classes('font-bold')
ui.label('This is my sample text')

ui.separator()

ui.markdown().default_classes('text-2xl')
ui.markdown('This is my sample text').classes('font-bold')
ui.markdown('This is my sample text')
```
Here the markdown elements are not picking up the text size by default. Can you please advise me on how to correctly use the default stylings for ui.markdown?

Screenshot here to compare:
![image](https://github.com/zauberzeug/nicegui/assets/65127974/56b53e1f-b108-42c2-a751-23fe4619b025)

Thanks a lot,
Anindya

**Comment by rodja** [comment, marked as answer]
Thanks for bringing this up @Anindya088. There was indeed a bug preventing the defaults from getting hold. I've fixed it in fe1cbd66ed9. But there are many other elements which do an assignment to  `self._classes` instead of extending the list. I'll need to discuss this with @falkoschindler.

**Reply by falkoschindler** [reply]
@rodja I guess we need to fix all these assignments to `self._classes`. Don't you think?

**Reply by rodja** [reply]
Yes!
